functions/Copy-SqlSsisCatalog.ps1

```diff
@@ -0,0 +1,472 @@
+                $envDestFolder.Refresh()
+            }
+            $srcEnv = ($sourceFolders | Where-Object { $_.Name -eq $Folder }).Environments[$Environment]
+            $targetEnv = New-Object "$ISNamespace.EnvironmentInfo" ($envDestFolder, $srcEnv.Name, $srcEnv.Description)
+            foreach ($var in $srcEnv.Variables) {
+                if ($var.Value.ToString() -eq "") { 
+                    $finalValue= ""
+                }
+                else {
+                    $finalValue = $var.Value
+                }
+                $targetEnv.Variables.Add($var.Name, $var.Type, $finalValue, $var.Sensitive, $var.Description)
+            }
+            Write-Output "Creating environment $Environment."
+            $targetEnv.Create()
+            $targetEnv.Alter()
+            $targetEnv.Refresh()
+        }
+
+        Function New-SSISDBCatalog {
+            param(
+                [System.Security.SecureString]$Password
+            )
+            if(!$Password) {
+                Write-Output "SSISDB Catalog requires a password."
+                $pass1 = Read-Host "Enter a password" -AsSecureString
+                $plainTextPass1 = [Runtime.InteropServices.Marshal]::PtrToStringAuto([Runtime.InteropServices.Marshal]::SecureStringToBSTR($pass1))
+                $pass2 = Read-Host "Re-enter password" -AsSecureString
+                $plainTextPass2 = [Runtime.InteropServices.Marshal]::PtrToStringAuto([Runtime.InteropServices.Marshal]::SecureStringToBSTR($pass2))
+                if ($plainTextPass1 -ne $plainTextPass2) {
+                    throw "Validation error, passwords entered do not match."
+                }
+                $plainTextPass = $plainTextPass1
+            }
+            else {
+                $plainTextPass = [Runtime.InteropServices.Marshal]::PtrToStringAuto([Runtime.InteropServices.Marshal]::SecureStringToBSTR($Password))
+            }
+
+            $catalog = New-Object "$ISNamespace.Catalog" ($destinationSSIS, "SSISDB", $plainTextPass)  
+            $catalog.Create()
+            $catalog.Refresh()
+        }
+        
+        $folder = $psboundparameters.Folder
+        $project = $psboundparameters.Project
+        $environment = $psboundparameters.Environment
+
+        $ISNamespace = "Microsoft.SqlServer.Management.IntegrationServices"
+
+        $sourceConnection = Connect-SqlServer -SqlServer $Source -SqlCredential $SourceSqlCredential
+        $destinationConnection = Connect-SqlServer -SqlServer $Destination -SqlCredential $DestinationSqlCredential
+
+        if ($sourceConnection.versionMajor -lt 11 -or $destinationConnection.versionMajor -lt 11) {
+            throw "SSISDB catalog is only available on Sql Server 2012 and above, exiting..."
+        }
+
+        try {
+            Get-RemoteIntegrationService -Computer $Destination
+        }
+        catch {
+            Write-Exception $_
+            throw "An error occured when checking the destination for Integration Services."
```

Comment:
I get this when IS isn't installed. Please add 

throw "An error occured when checking the destination for Integration Services. Is Integration Services installed?"